### ISSUE TITLE
put_wall_post not working now

### ISSUE THREAD

**Issue by skipcloud** [arsduo/koala#517]
Figured I'd open an issue as I've been messing around this with gem today to try create something silly on Facebook, tried to post to a friends wall and got this error

`Feed story publishing to other users is disabled for this application, x-fb-trace-id: EZaMJ3t2G8m [HTTP 403] (Koala::Facebook::ClientError)`

apparently Facebook removed the ability for apps to post to walls via the Graph API [in 2013](https://developers.facebook.com/docs/apps/migrations/completed-changes#feb_2013)

**Comment by aaackerman** [comment]
Hey @skipcloud, I haven't tried your use case specifically, but I wanted to note that the `put_wall_post` method can be used for different types of graphs. For example, it still works for posting to Facebook pages using a page token.

**Comment by skipcloud** [comment]
@aaackerman yeah when I used it I could post to my own wall, but posting to other people's wall is a no-go apparently. I guess the issue would be updating the docs to show that.

**Comment by aaackerman** [comment]
@skipcloud  I tested a few moments ago by passing the user's access token to `Koala::Facebook::API.new` and posting to their wall using `put_wall_post`. Everything worked as expected, leaving a post to the user's page. Are you using oauth to get the token as well as requesting the `publish_actions` [permission](https://developers.facebook.com/docs/facebook-login/permissions) on your Facebook app? That may solve the issue, as you will not be able to post without that permission and token. To have this work on non-test users, you may have to get your app reviewed and approved by facebook for the permission. 

```
user_graph = Koala::Facebook::API.new(auth.token)
user_graph.put_wall_post("Test message from api")
 => returns id of post that was added to the user's wall
```

**Comment by skipcloud** [comment]
@aaackerman yes I used oauth to sign in, and I was able to post to my own wall. When I used the app to try post something to a friends wall I was not able not able to do so. The link in my original post shows when Facebook removed the ability to post to friends walls a while back for applications.

**Comment by aaackerman** [comment]
@skipcloud, gotcha. If your friend is not listed in your Facebook app under the 'Roles' page as an administrator, developer, or tester, the post will not work because your app may not be approved for public use of the `publish_actions` feature. If you check out the `publish_actions` header in the link I added above, you will see you can still add Posts on a user's behalf (not quite the same feature described in your link). It works for your account because you are probably an admin of the app.  Some clarity about how Facebook permissions work could be a nice addition to the docs, will look into contributing! Hope this helps.

**Comment by arsduo** [comment]
Thanks, both of you! I'll review the docs and update them appropriately if any of them say you can do that.
